value-fixture/src/org/immutables/fixture/builder/FirstPartyImmutableWithDifferentStyle.java

```diff
@@ -0,0 +1,16 @@
+package org.immutables.fixture.builder;
+
+import org.immutables.value.Value.Immutable;
+import org.immutables.value.Value.Style;
+
+@Immutable
+@Style(
+    builder = "getTheBuilder",
+    build = "doIIT",
+    from = "makeDaCopy",
+    typeBuilder = "Abonabon"
+)
+public abstract class FirstPartyImmutableWithDifferentStyle {
```

Comment:
In theory, if style would allow auto-discovery, then we would just detect how particular type's is integrated. I cannot be 100% sure about that, but it seems that we are losing very little if anything if we not using precise configuration (overridable on per-attribute level) for custom builders. The general style would allow number of naming patterns, and auto-discovery would use those patterns along with checking parameter and return types to figure out how particular builder is integrated.